### ISSUE TITLE
RCS1208 False-positive on flat if statement with multiple conditions

### ISSUE THREAD

**Issue by wiciok** [dotnet/roslynator#1454]
**Product and Version Used**:
Roslynator.Analyzers 4.12.2

**Steps to Reproduce:**
Following code or similar one containing if statement with multiple conditions, but without any nesting:
```csharp
private void Foo(string bar, int baz)
{
    if (bar == "bar" && baz == 123)  // RCS1208: Reduce 'if' nesting
    {
        var foo = "baz";
    }
}
```
Please note, that the issue doesn't occur, when there is some other code present after if statement in question:

```csharp
private void Foo(string bar, int baz)
{
    if (bar == "bar" && baz == 123)
    {
        var foo = "baz";
    }
    var thisDoes = "something preventing RCS1208 in if statement above";
}
```

**Actual Behavior**:

RCS1208 is raised, suggesting rather counterproductive change to:
```csharp
private void Foo(string bar, int baz)
{
    if (bar != "bar" || baz != 123)
    {
        return;
    }
    var foo = "baz";
}

```
**Expected Behavior**:
Do not raise RCS1208 in case of flat if statements with multiple conditions.

**Comment by AdrianoAE** [comment]
Also facing false positive, this triggers the analyzer:
```csharp
if (last.End > time)
{
    return last;
}

return null;
```

but if the if is inverted:
```csharp
if (last.End <= time)
{
    return null;
}

return last;
```

then it's fine

**Comment by wouter-m-palmsens** [comment]
I have 
```
if (runtimeParameters.MeasureForwardAndReverseCurrent)
{
    p.Add(forwardVariable).Add(reverseVariable);
}
```
changed to 
```
if (!runtimeParameters.MeasureForwardAndReverseCurrent)
{
    return;
}

p.Add(forwardVariable).Add(reverseVariable);
```

Not only is there no reduction of 'if' nesting, but to me, the new code looks ugly.
